docs/source/developer/tests.rst

```diff
@@ -39,67 +39,143 @@ If you need access to Windows you can download a
 `virtual machine <https://developer.microsoft.com/en-us/windows/downloads/virtual-machines>`_
 directly from Microsoft for any hypervisor.
 
-Decorating tests
-~~~~~~~~~~~~~~~~
 
-The ``functional`` decorator marks tests that don't require any pre-existing
-AWS context, and can therefore be run cleanly against live AWS.
+Writing Tests for Cloud Controlled Resources
+--------------------------------------------
 
-To run only the tests decorated by ``functional``:
+Cloud Custodian makes use of flight recording to allow for both functional and unit testing.
+Several libraries are used to perform flight recording and replay.
```

Comment:
would switch this line to 

Each of the provider's uses a separate technique that integrates with the provider sdk to handle flight recording of custodian's api calls. For setting up test scenarios and infrastructure provisioning we use pytest-terraform.

please omit the call out on placebo, its aws specific and by that logic we would have the other provider sdk specific libraries called out. its more important to identify the pattern and our abstraction (record_flight/replay_flight) wrt to custodian api calls, and pytest-terraform for infra setup.